src/modules/flow/netctl/netctl.c

```diff
@@ -0,0 +1,336 @@
+/*
+ * This file is part of the Soletta™ Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "sol-flow/netctl.h"
+#include "sol-flow-internal.h"
+
+#include <sol-netctl.h>
+#include <sol-str-table.h>
+#include <sol-util-internal.h>
+#include <errno.h>
+
+struct network_service_data {
+    char *service_name;
+    enum sol_netctl_service_state state;
+};
+
+void
+service_list_cb(void *data, const struct sol_netctl_service *service)
+{
+    struct sol_flow_node *node = data;
+    const char *name;
+    int r;
+
+    name = sol_netctl_service_get_name(service);
+
+    if (sol_netctl_service_get_state(service) ==
+        SOL_NETCTL_SERVICE_STATE_REMOVE) {
+        r = sol_flow_send_string_packet(node,
+            SOL_FLOW_NODE_TYPE_NETCTL_LIST_SERVICES__OUT__REMOVED,
+            name);
+        if (r < 0)
+            SOL_WRN("Failed to send removed service name: %s", name);
+        return;
+    }
+
+    r = sol_flow_send_string_packet(node,
+        SOL_FLOW_NODE_TYPE_NETCTL_LIST_SERVICES__OUT__ADDED,
+        name);
+    if (r < 0)
+        SOL_WRN("Failed to send added service name: %s", name);
+}
+
+static int
+open_list_services(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    int r;
+
+    r = sol_netctl_add_service_monitor(service_list_cb, node);
+    SOL_INT_CHECK(r, < 0, r);
+
+    return 0;
+}
+
+static void
+close_list_services(struct sol_flow_node *node, void *data)
+{
+    static const char *errmsg = "Could not delete service monitor";
+
+    if (sol_netctl_del_service_monitor(service_list_cb, node))
+        SOL_WRN("%s", errmsg);
+}
+
+void
+service_status_cb(void *data, const struct sol_netctl_service *service)
+{
+    struct sol_flow_node *node = data;
+    struct network_service_data *mdata = sol_flow_node_get_private_data(node);
+    enum sol_netctl_service_state current_state;
+    const char *name;
+    int r;
+
+    name = sol_netctl_service_get_name(service);
+
+    if (!name || strcmp(name, mdata->service_name)) 
+        return;
+
+    current_state = sol_netctl_service_get_state(service);
+    if(current_state != mdata->state) {
```

Comment:
like above, you can do an early return here:

```c
if (current_state == mdata->state)
    return;
```

Also check the style, it's missing a space. Please run `./tools/check-style.py` or even better, use a GIT hook so you don't forget it: https://github.com/solettaproject/soletta/wiki/Contributing#git-hooks